test/e2e/serial/tests/sched_cache.go

```diff
@@ -0,0 +1,512 @@
+
+				klog.Infof("Creating %d pods each requiring %q", desiredPods, e2ereslist.ToString(podRequiredRes))
+				for _, testPod := range testPods {
+					err := fxt.Client.Create(context.TODO(), testPod)
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				// note the cleanup is done automatically once the ns on which we run is deleted - the fixture takes care
+
+				// very generous timeout here. It's hard and racy to check we had 2 pods pending (expected phased scheduling),
+				// but that would be the most correct and stricter testing.
+
+				failedPods, updatedPods := wait.ForPodListAllRunning(fxt.Client, testPods, 180*time.Second)
+				if len(failedPods) > 0 {
+					nrtListFailed, _ := e2enrt.GetUpdated(fxt.Client, nrtv1alpha1.NodeResourceTopologyList{}, time.Minute)
+					klog.Infof("%s", e2enrtint.ListToString(nrtListFailed.Items, "post failure"))
+
+					for _, failedPod := range failedPods {
+						_ = objects.LogEventsForPod(fxt.K8sClient, failedPod.Namespace, failedPod.Name)
+					}
+				}
+				Expect(len(failedPods)).To(BeZero(), "unexpected failed pods: %q", accumulatePodNamespacedNames(failedPods))
+
+				for _, updatedPod := range updatedPods {
+					schedOK, err := nrosched.CheckPODWasScheduledWith(fxt.K8sClient, updatedPod.Namespace, updatedPod.Name, serialconfig.Config.SchedulerName)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(schedOK).To(BeTrue(), "pod %s/%s not scheduled with expected scheduler %s", updatedPod.Namespace, updatedPod.Name, serialconfig.Config.SchedulerName)
+				}
+			})
+
+			It("should keep non-fitting pod in pending state forever", func() {
+
+				hostsRequired := 2
+				NUMAZonesRequired := 2
+				desiredPods := 3
+
+				// we have 2 compute nodes each with 2 numa zones. A resource (easiest is devices, thus devices) is available
+				// only on 1 out of N (=2) NUMA zones.
+				// So if we run M>2 pods each requiring > 50 % cpus on each NUMA zone, and 1 device each, we must get
+				// 2 running pods and M-2 pending pods.
+				// In this testcase, having running pods is not good enough: we want to ALSO have pods which keep being
+				// pending "forever". We can't really check "forever", so we just check "long enough".
+
+				deviceName := e2efixture.GetDeviceType1Name()
+				if deviceName == "" {
+					e2efixture.Skipf(fxt, "missing required device name (device1)")
+				}
+
+				Expect(desiredPods).To(BeNumerically(">", hostsRequired)) // this is more like a C assert. Should never ever fail.
+
+				expectedPending := desiredPods - hostsRequired
+				klog.Infof("hosts required %d desired pods %d expected pending %d", hostsRequired, desiredPods, expectedPending)
+
+				// so we can't support ATM zones > 2. HW with zones > 2 is rare anyway, so not to big of a deal now.
+				By(fmt.Sprintf("filtering available nodes with at least %d NUMA zones", NUMAZonesRequired))
+				nrtCandidates = e2enrt.FilterZoneCountEqual(nrtList.Items, NUMAZonesRequired)
```

Comment:
Similar to one of my comments above, lets add a TODO here to move to `FilterZoneCountAtLeast`.

Reply:
done!